pilot/pkg/xds/xds_cache.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"sync"
+
+	"github.com/golang/protobuf/ptypes/any"
+
+	"istio.io/istio/pkg/config/host"
+)
+
+// Cache interface defines a store for caching XDS responses
+// Note this is currently only for EDS, and will need some modifications to support other types
+// All operations are thread safe
+type Cache interface {
+	Insert(key EndpointBuilder, value *any.Any)
+	Get(key EndpointBuilder) (*any.Any, bool)
+	Clear(edsUpdatedServices map[string]struct{})
+	ClearAll()
+}
+
+type inMemoryCache struct {
+	store map[host.Name]map[string]*any.Any
```

Comment:
Any case where we would have a 'split view' - due to different Sidecar importTo ? I.e. same hostname returning different set of endpoints depending on caller ?

Reply:
Yes, but its part of the key. See updated code, the key is `string(b.service.Hostname)+"/"+b.service.Attributes.Namespace`.